runbotics/runbotics-desktop/src/action/zip/zip.test.ts

```diff
@@ -0,0 +1,282 @@
+        });
+
+        it('Should create zip named with new zipName when optional path is not provided', async() => {
+            const fileName = 'toArchiveName';
+            const zipName = 'zippedName';
+            const params: ZipFileActionInput = {
+                fileName,
+                path: undefined,
+                zipName 
+            };
+
+            createTestFolder(`${TEMP_FOLDER_PATH}${path.sep}${fileName}`);
+            await zipActionHandler.zipFile(params);
+            expect(
+                fs.existsSync(`${TEMP_FOLDER_PATH}${path.sep}${zipName}.zip`)
+            ).toBeTruthy();
+            removeTestFolder(`${TEMP_FOLDER_PATH}${path.sep}${fileName}`);
+            removeTestFolder(`${TEMP_FOLDER_PATH}${path.sep}${zipName}.zip`);
+        });
+
+        it('Should create zip named with new zipName when all parameters are provided', async() => {
+            const fileName = 'toArchiveName';
+            const toZipPath = `${TEMP_FOLDER_PATH}${path.sep}${fileName}`;
+            const zipName = 'zippedName';
+            const params: ZipFileActionInput = {
+                fileName,
+                path: TEMP_FOLDER_PATH,
+                zipName
+
+            };
+
+            createTestFolder(toZipPath);
+            await zipActionHandler.zipFile(params);
+            expect(
+                fs.existsSync(`${TEMP_FOLDER_PATH}${path.sep}${zipName}.zip`)
+            ).toBeTruthy();
+            removeTestFolder(toZipPath);
+            removeTestFolder(`${TEMP_FOLDER_PATH}${path.sep}${zipName}.zip`);
+        });
+
+        it('Should throw error when fileName is not found', async() => {
+            const params: ZipFileActionInput = {
+                fileName: 'notExistingFolderName',
+                path: undefined,
+                zipName: undefined
+            };
+
+            await expect(zipActionHandler.zipFile(params)).rejects.toThrowError(
+                'Create archive: no such file or directory'
+            );
+        });
+
+        it('Should throw error when fileName is not provided', async() => {
+            const params: ZipFileActionInput = {
+                fileName: undefined,
+                path: undefined,
+                zipName: undefined
+            };
+
+            await expect(zipActionHandler.zipFile(params)).rejects.toThrowError(
+                'File name is mandatory'
+            );
+        });
+
+        it('Should throw error when zip file already exists', async() => {
+            const fileName = 'toArchive';
+            const params: ZipFileActionInput = {
+                fileName,
+                path: undefined,
+                zipName: undefined
+            };
+
+            createTestZip(fileName, TEMP_FOLDER_PATH);
+            await expect(zipActionHandler.zipFile(params)).rejects.toThrowError(
+                'Create archive action: File with this name already exists.'
```

Comment:
If someone changes error message and forgets it here the tests won't pass, right?

maybe an enum ZipErrorMessages?

Reply:
this enum keys can be even error codes - e.g. 
```typescript
enum ZipErrorMessages {
EBUSY: "File with this name already exists."
}
``` 
so u can replace the whole switch with just:
 throw new Error(`${actionName} action: ${ZipErrorMessages[errorCode]}`)